### PR TITLE
feat(core): scope name within the same resource should be unique

### PR DESCRIPTION
**Pull request by wangsijie**
## Summary

scope name within the same resource should be unique

## Testing

UTs

### CODE REVIEW

File: packages/core/src/routes/resource.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import { Resources, Scopes } from '@logto/schemas';
 import { object, string } from 'zod';
 
 import { isTrue } from '#src/env-set/parameters.js';
+import RequestError from '#src/errors/RequestError/index.js';
```
Comment (gao-sun) [inline packages/core/src/routes/resource.ts:6]: would you like to add some integration tests to cover this case?

Comment (wangsijie) [reply packages/core/src/routes/resource.ts:6]: Good call, it do remind me about it

File: packages/phrases/src/locales/en/errors.ts [hunk 2]
```diff
@@ -180,6 +180,9 @@ const errors = {
     scope_exists: 'The scope id {{scopeId}} has already been added to this role',
     user_exists: 'The user id {{userId}} is already been added to this role',
```
Comment (charIeszhao) [inline packages/phrases/src/locales/en/errors.ts:181]: nit: might be a little bit unrelated to this PR, but these should be revised. `is already been` is grammarly incorrect

**Review by charIeszhao — APPROVED** [review]
Overall looks good. Would be great if you can add a test case for the new query

**Review by simeng-li — APPROVED** [review]
LGTM

**Pull request merged**